go/vt/vtgate/vcursor_impl.go

```diff
@@ -706,6 +722,31 @@ func (vc *vcursorImpl) KeyspaceAvailable(ks string) bool {
 	return exists
 }
 
+// ErrorIfShardedF implements the VCursor interface
+func (vc *vcursorImpl) ErrorIfShardedF(ks *vindexes.Keyspace, warn, errFormat string, params ...interface{}) error {
+	if ks.Sharded {
+		return vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, errFormat, params...)
+	}
+	if vc.warnShardedOnly {
+		vc.warnings = append(vc.warnings, &querypb.QueryWarning{
+			Code:    mysql.ERNotSupportedYet,
+			Message: fmt.Sprintf("use of feature that is only supported in unsharded mode: %s", warn),
```

Comment:
can we change the error message to `'%s' not supported in sharded mode`

Reply:
👍